cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java

```diff
@@ -71,7 +71,8 @@ protected TSStatus executeNonQueryPlan(PhysicalPlan plan) {
       if (!(plan instanceof SetSystemModePlan)
           && !(plan instanceof FlushPlan)
           && IoTDBDescriptor.getInstance().getConfig().isReadOnly()) {
-        throw new QueryProcessException(
+        return RpcUtils.getStatus(
+            TSStatusCode.QUERY_PROCESS_ERROR,
             "Current system mode is read-only, does not support non-query operation");
```

Comment:
I just feel a little bit weird why a NonQueryPlan throws a  QUERY_PROCESS_ERROR...
Should we change it to another error code?

Reply:
> I just feel a little bit weird why a NonQueryPlan throws a QUERY_PROCESS_ERROR... Should we change it to another error code?

It makes sense. Is `WRITE_PROCESS_REJECT(413)` OK?

Reply:
How about the `READ_ONLY_SYSTEM_ERROR(502)`?

Reply:
> How about the `READ_ONLY_SYSTEM_ERROR(502)`?

Althogh the name is reasonable, the description is not as our expected.
![image](https://user-images.githubusercontent.com/3821212/146319212-1f43ec07-583c-4358-bc52-46a8364d1379.png)

Reply:
> Althogh the name is reasonable, the description is not as our expected. ![image](https://user-images.githubusercontent.com/3821212/146319212-1f43ec07-583c-4358-bc52-46a8364d1379.png)

Is there any problem about the description?

Reply:
I haven't found any usage of READ_ONLY_SYSTEM_ERROR yet. Maybe we can change the sentence `Operating system is read only` to `database is read only`? The 'Opertating system' is confused with our common word `OS`(Unix or Windows).

Reply:
> I haven't found any usage of READ_ONLY_SYSTEM_ERROR yet. Maybe we can change the sentence `Operating system is read only` to `database is read only`? The 'Opertating system' is confused with our common word `OS`(Unix or Windows).

Makes sense. Yes, `database is read only` looks good to me.

Reply:
> > I haven't found any usage of READ_ONLY_SYSTEM_ERROR yet. Maybe we can change the sentence `Operating system is read only` to `database is read only`? The 'Opertating system' is confused with our common word `OS`(Unix or Windows).
> 
> Makes sense. Yes, `database is read only` looks good to me.

I refine the codes and add a new exception class. Please take a review again.